tests/Benchmarks/System.Buffers/ToSpan.cs

```diff
@@ -0,0 +1,35 @@
+﻿using System;
+using System.Buffers;
+using System.Buffers.Tests;
+using System.Collections.Generic;
+using System.Collections.Sequences;
+using System.Linq;
+using System.Text;
+using BenchmarkDotNet.Attributes;
+
+namespace Benchmarks.System.Buffers
+{
+    public class ToSpan
+    {
+        public static ReadOnlySequence<byte> Sequence = BufferFactory.Create(Enumerable.Repeat(0, 10000000)
+            .Select(x => new Memory<byte>(new byte[] { 0 })));
```

Comment:
You measured `ROS<byte>` with 1_024 segments each of size 1. That isn't going to be common in practice. I would recommend improving the test cases in this benchmark by including more realistic permutations of `ROS<byte>` with 16, 64, 512 segments of size 1_024, 4096 bytes (leaving the ResizableArray starting size fixed at 1_024).

I am confident the new approach in your PR will be faster regardless (even more so if you use the constant time `Length` property rather than calculating it yourself), but having such details within the benchmark will be helpful.